### PR TITLE
Task/dr 559 implement lambda powertools

### PR DESCRIPTION
**Pull request by Pira-nhs**
## Description

Implementation of AWS Lambda Power Tools 

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [x] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: scripts/app-build-deploy.sh [hunk 1]
```diff
@@ -16,21 +16,13 @@ echo "Service Name Workspace: ${SERVICE_NAME_WORKPACE}"
 
 cd ./application/${SERVICE_NAME}
 
-rsync -Rr ./ ./chalicebuildtmp
-cd ./chalicebuildtmp
-cp -r ../../../application-utils/* ./chalicelib/
-pip install -r requirements.txt --target .
+zip -r deployment.zip .
```
Comment (timrickwood) [inline scripts/app-build-deploy.sh:19]: Above we have identical copies of the utitlities.py per lambda - which will trip our code duplication quality gate. The true source is held once here - application-utils/common/utilities.py and copied into the relevant lambda folder (both here as part of deployment and as part of unit testing). I think we need to keep the line that copies the application-utils folder before we zip up the deployment package

File: infrastructure/stacks/application/lambda.tf [hunk 2]
```diff
@@ -63,6 +64,7 @@ module "organisation-affiliation-data-manager-lambda" {
 
   function_name = var.organisation_affiliations_function_name
   description   = "Microservice for interacting with organisation affiliation dynamodb table"
+  layers        = ["arn:aws:lambda:eu-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:46"]
```
Comment (timrickwood) [inline infrastructure/stacks/application/lambda.tf:67]: move up to module ?

File: infrastructure/stacks/application/lambda.tf [hunk 3]
```diff
@@ -10,6 +10,7 @@ module "organisations-data-manager-lambda" {
 
   function_name = var.organisations_function_name
   description   = "Microservice for interacting with organisations dynamodb table"
+  # layers        = ["arn:aws:lambda:eu-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:46"]
```
Comment (tamaragoldschmidt) [inline infrastructure/stacks/application/lambda.tf:13]: Does this still need to be here now you've moved it to the module?

File: infrastructure/stacks/application/lambda.tf [hunk 4]
```diff
@@ -116,6 +118,7 @@ module "locations-data-manager-lambda" {
 
   function_name = var.locations_function_name
   description   = "Microservice for interacting with locations dynamodb table"
+  # layers        = ["arn:aws:lambda:eu-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:46"]
```
Comment (tamaragoldschmidt) [inline infrastructure/stacks/application/lambda.tf:121]: Does this still need to be here now you've moved it to the module?

File: infrastructure/stacks/application/lambda.tf [hunk 5]
```diff
@@ -63,6 +64,7 @@ module "organisation-affiliation-data-manager-lambda" {
 
   function_name = var.organisation_affiliations_function_name
   description   = "Microservice for interacting with organisation affiliation dynamodb table"
+  # layers        = ["arn:aws:lambda:eu-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:46"]
```
Comment (tamaragoldschmidt) [inline infrastructure/stacks/application/lambda.tf:67]: Does this still need to be here now you've moved it to the module?

File: infrastructure/stacks/application/lambda.tf [hunk 6]
```diff
@@ -169,6 +172,7 @@ module "healthcare-services-data-manager-lambda" {
 
   function_name = var.healthcare_services_function_name
   description   = "Microservice for interacting with healthcare services dynamodb table"
+  # layers        = ["arn:aws:lambda:eu-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:47"]
```
Comment (tamaragoldschmidt) [inline infrastructure/stacks/application/lambda.tf:175]: Does this still need to be here now you've moved it to the module?

File: infrastructure/stacks/application/lambda.tf [hunk 7]
```diff
@@ -222,6 +226,7 @@ module "questionnaire-data-manager-lambda" {
 
   function_name = var.questionnaires_function_name
   description   = "Microservice for interacting with questionnaire dynamodb table"
+  # layers        = ["arn:aws:lambda:eu-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:46"]
```
Comment (tamaragoldschmidt) [inline infrastructure/stacks/application/lambda.tf:229]: Does this still need to be here now you've moved it to the module?

File: infrastructure/stacks/application/lambda.tf [hunk 8]
```diff
@@ -275,6 +280,7 @@ module "questionnaire-response-data-manager-lambda" {
 
   function_name = var.questionnaire_responses_function_name
   description   = "Microservice for interacting with questionnaire response dynamodb table"
+  # layers        = ["arn:aws:lambda:eu-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:46"]
```
Comment (tamaragoldschmidt) [inline infrastructure/stacks/application/lambda.tf:283]: Does this still need to be here now you've moved it to the module?

File: infrastructure/stacks/application/lambda.tf (hunk 3 above)
Comment (timrickwood) [reply infrastructure/stacks/application/lambda.tf:13]: good spot . No but i think I have written it so you can add specific layers from the stack if needed. I will remove htis. dead code - thanks @tamaragoldschmidt

**Review by tamaragoldschmidt — APPROVED** [review]

**Review by jope1 — APPROVED** [review]

**Pull request merged**